roles/ceph-validate/tasks/check_rgw_pools.yml

```diff
@@ -0,0 +1,44 @@
+---
+- name: fail if pool_type is not set
```

Comment:
I think this is too restrictive and some tasks are breaking the backward compatibility.

As I mentioned in a previous comment, users that wouldn't have defined pool_type in the rgw_create_pools dict will have the playbook failing.

Some checks make sense, like checking ec_m/ec_k when the pool type is ec but others like size and pg_num have default value so you don't actually need to always define these keys in the dict (ie: `{{ item.value.pg_num | default(osd_pool_default_pg_num) }}`)